controllers/metal3remediation_controller.go

```diff
@@ -141,76 +149,96 @@ func (r *Metal3RemediationReconciler) reconcileNormal(ctx context.Context,
 	}
 
 	if remediationType == infrav1.RebootRemediationStrategy {
-		// If no phase set, default to running
+		// If no phase set, default to running and set time and retry count
 		if remediationMgr.GetRemediationPhase() == "" {
 			remediationMgr.SetRemediationPhase(infrav1.PhaseRunning)
+			now := metav1.Now()
+			remediationMgr.SetLastRemediationTime(&now)
+			remediationMgr.IncreaseRetryCount()
+			return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
+		}
+
+		// try to get node
+		node, err := remediationMgr.GetNode(ctx)
```

Comment:
It looks to me like `getClusterClient()` is potentially getting a client for another cluster.
Having this here means that e.g. if the Node is in another cluster and we don't have RBAC access to it (wasn't required before) then we won't be able to do _anything_ to remediate. Is it possible to avoid this error up to the point where we actually need the Node?

Reply:
done

Reply:
do I understand correctly, it can be that capm3 is updated, but updated RBAC roles won't apply for existing clusters?

Reply:
I'm handling the error now by skipping node releated actions